src/s2/s2region_term_indexer.cc

```diff
@@ -61,6 +61,12 @@
 //   never be any document regions larger than the query region.  This can
 //   significantly reduce the size of queries.
 //
+// + If the query will contain only points (rather than general regions), then
+//   we can skip all the ancestor terms mentioned above (except last cell see
+//   `GetIndexTerms(const S2Point& point...` for details) because there will
+//   never be any document regions larger than the index region.  This can
```

Comment:
I think saying "never be any document regions smaller than the query region" makes the comment have the same sense as 1. above.

Reply:
you're right

Reply:
> I already wrote, now it's ok for me don't have such api for query, but for indexing, calling other api make my code faster (a little but measurable), so I want other api.

Do you have an indexing benchmark, or are you referring to the microbenchmarks you wrote in #304? What are the results?

> Offtop:
> Btw today we discussed how speedup geo terms search: now we need to make disjunction of all query terms (it's from s2 documentation and it works now for us).

As you say, off-topic for this PR. Please open a discussion or an issue.

Reply:
@jmr Today another day when I thought that will be good to have such API

But because now you don't have it I need to write code like this:
```cpp
    geoTerms = _indexer.GetIndexTerms(*_shape.region(), {});
    if (!_shape.contains(centroid)) {
      auto terms = _indexer.GetIndexTerms(centroid, {});
      geoTerms.insert(geoTerms.end(), std::make_move_iterator(terms.begin()),
                      std::make_move_iterator(terms.end()));
```

Reply:
> Do you have an indexing benchmark, or are you referring to the microbenchmarks you wrote in https://github.com/google/s2geometry/pull/304? What are the results?

We already computed it save at least N*10 ns, where N number of inserts.
Why do you against this API so much?